src/bin/josh-filter.rs

```diff
@@ -113,11 +124,53 @@ fn run_filter(args: Vec<String>) -> josh::JoshResult<i32> {
         josh::apply_filter_to_refs(
             &repo,
             &*viewobj,
-            &[(src.clone(), t)],
+            &[(src.clone(), t.clone())],
             &mut fm,
             &mut backward_maps.write().unwrap(),
         );
 
+        let mut all_dirs = vec![];
+
+        if check_permissions {
+            let result_tree = repo.find_reference(&t)?.peel_to_tree()?;
+
+            result_tree.walk(git2::TreeWalkMode::PreOrder, |_, entry| {
+                let name = entry.name().unwrap();
```

Comment:
You leave a lot of unwrap()s. Is it fine to have it here? Could entry possibly be wrong? Why not unwrap_or()?

Reply:
This is a command line tool, not a long running (server) process. So leaving some unwrap is not such a big problem here. Especially for things we don't expect to ever happen.